src/steamship/agents/mixins/transports/telegram.py

```diff
@@ -74,31 +82,38 @@ def _instance_init(self):
 
     @post("telegram_webhook_info")
     def telegram_webhook_info(self) -> dict:
-        return requests.get(self.api_root + "/getWebhookInfo").json()
-
-    @post("connect_telegram")
-    def connect_telegram(self, bot_token: str):
-        self.store.set("bot_token", {"token": bot_token})
-        self.bot_token = bot_token
+        api_root = self.get_api_root()
+        if not api_root:
+            raise SteamshipError(
+                message="Unable to fetch Telegram API info -- perhaps your bot token isn't set?"
+            )
 
-        try:
-            self.instance_init()
-            return "OK"
-        except Exception as e:
-            return f"Could not set webhook for bot. Exception: {e}"
+        return requests.get(api_root + "/getWebhookInfo").json()
 
     @post("telegram_disconnect_webhook")
     def telegram_disconnect_webhook(self, *args, **kwargs):
         """Unsubscribe from Telegram updates."""
-        requests.get(f"{self.api_root}/deleteWebhook")
+        api_root = self.get_api_root()
+        if not api_root:
+            raise SteamshipError(
+                message="Unable to disconnect from Telegram -- perhaps your bot token isn't set?"
+            )
```

Comment:
Obviously a non-blocker, but if you put this chunk in `get_api_root()` you wouldn't have to repeat it so much.